### ISSUE TITLE
With navigation.indexes, nav doesn't recognise dirs with no other pages

### ISSUE THREAD

**Issue by jaklan** [squidfunk/mkdocs-material#3071]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Let's say we have a following structure: `docs/projects/project-name/docs`. In `docs` folder we only have an `index.md` file. If we enable `navigation.indexes` it works as expected, but we still see an arrow on the right side, which indicates there are some pages within given section, which is not true - it's just empty.

### Expected behaviour

The expected behaviour is already implemented in the `mkdocs-section-index` plugin - in such scenario there's no arrow on the right side.
![image](https://user-images.githubusercontent.com/24907857/135720212-2a59e27b-a9a3-495d-ace2-6c3dd183657d.png)

### Actual behaviour

![image](https://user-images.githubusercontent.com/24907857/135719916-90340a11-5fee-4839-9d60-390c9f212a2d.png)

### Steps to reproduce

1. Create `docs/projects/project-name/docs` dirs.
2. Create only `index.md` file in the last dir.
3. Enable `navigation.indexes`.
4. Run `mkdocs serve`.

### Package versions

- Python: 3.8.12
- MkDocs: 1.2.2
- Material: 7.3.1

### Configuration

```yaml
site_name: Whatever

theme:
  name: material
  features:
    - navigation.tabs
    - navigation.top
    - navigation.indexes
```

### System information

- Operating system: macOS
- Browser: Chrome

**Comment by squidfunk** [comment]
Thanks for reporting 😊 Fixed in 149b0dbc4.

**Comment by jaklan** [comment]
That was fast 😄 Thanks!
